### PR TITLE
Add tests for LEB128 unsigned 32-bit decoder

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: test/unittests/leb128_test.cpp [hunk 1]
```diff
@@ -66,6 +66,47 @@ TEST(leb128, decode_u64_invalid)
...
+    // clang-format off
+    std::vector<std::pair<bytes, uint32_t>> testcases = {
+        {{0}, 0}, // 0
+        {{0x80, 0x80, 0x00}, 0}, // 0 with leading zeroes
+        {{1}, 1}, // 1
+        {{0x81, 0x80, 0x80, 0x00}, 1}, // 1 with leading zeroes
+        {{0x81, 0x80, 0x80, 0x80, 0x00}, 1}, // 1 with max leading zeroes
+        {{0xe5, 0x8e, 0x26}, 624485}, // 624485
+        {{0xe5, 0x8e, 0xa6, 0x80, 0x00}, 624485}, // 624485 with leading zeroes
+        {{0xff, 0xff, 0xff, 0xff, 0x07}, 0x7fffffff},
```
Comment (axic) [inline test/unittests/leb128_test.cpp]: Can you also add a test case for the value `0x80000000` (encoded as `8080808008`) ? Also to u64 perhaps?

Comment (gumb0) [reply test/unittests/leb128_test.cpp]: added

File: test/unittests/leb128_test.cpp [hunk 2]
```diff
@@ -33,6 +33,8 @@ TEST(leb128, decode_u64)
         {{0x81, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x00}, 1}, // 1 with max leading zeroes
         {{0xe5, 0x8e, 0x26}, 624485}, // 624485
         {{0xe5, 0x8e, 0xa6, 0x80, 0x80, 0x00}, 624485}, // 624485 with leading zeroes
+        {{0xff, 0xff, 0xff, 0xff, 0x07}, 0x7fffffff},
+        {{0x80, 0x80, 0x80, 0x80, 0x08}, 0x80000000},
```
Comment (axic) [inline test/unittests/leb128_test.cpp]: Can you also add 64-bit versions of these?

Comment (gumb0) [reply test/unittests/leb128_test.cpp]: added

**Pull request merged**